File: 2023/02/part1.js
const fs = require("fs");

if (process.argv.lenght < 3) {
  console.log("please specify input file")
  process.exit(1);
}

const colours_max = {"red": 12, "green": 13, "blue": 14};
const data = fs.readFileSync(process.argv[2]).toString().split("\n").filter(i => i != "")
let total = 0;
outer:
for (let line of data) {
  [p1, p2] = line.split(":");
  [_, gameNumber] = p1.split(" ");
  for (const [_, num, colour] of line.split(":").pop().matchAll(/([0-9]+)\s(blue|green|red)(,|;|$)/g)) {
    if (num > colours_max[colour]) {
      continue outer;
    }
  }
  total += parseInt(gameNumber);
}
console.log(total);

File: 2023/02/part1.py
import sys

if len(sys.argv) < 2:
    print("please specify input file")
    sys.exit(1)

f = open(sys.argv[1], 'r')
colour_max = {"red": 12, "green": 13, "blue": 14}

sumOfValids = 0
for line in f.read().splitlines():
    res = line.split(":")
    gameNumber = res[0].replace("Game ", "")
    sets = res[1].split(";")
    invalidSet = False
    for s in sets:
        items = list(s.split(","))
        for i in items:
            [_, num, colour] = i.split(" ")
            if int(num) > colour_max[colour]:
                invalidSet = True
                break
    if invalidSet:
        continue

    sumOfValids += int(gameNumber)

print(sumOfValids)
f.close()
